From 7e4afe9c7995510a80ac02d684b6cb1ffc653765 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Fri, 19 Apr 2024 15:01:08 +0200
Subject: [PATCH] arm64: dts: st: remove USB33 from stm32mp25

Remove PWR USB33 that has been suppressed from the ref manual.
USB vdd33-supply is directly supplied by the PMIC LDO4 via
scmi_vdd3v3_usb.

Change-Id: I48bbe370fcc0aadf7929a42b975277a9ce8ab51a
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/365099
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/375133
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi     | 6 ------
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts  | 6 ++++++
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 6 ++++++
 3 files changed, 12 insertions(+), 6 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -148,10 +148,6 @@
 						reg = <VOLTD_SCMI_UCPD>;
 						regulator-name = "vdd33ucpd";
 					};
-					scmi_vdd33usb: regulator@6 {
-						reg = <VOLTD_SCMI_USB33>;
-						regulator-name = "vdd33usb";
-					};
 					scmi_vdda18adc: regulator@7 {
 						reg = <VOLTD_SCMI_ADC>;
 						regulator-name = "vdda18adc";
@@ -266,7 +262,6 @@
 		st,syscfg       = <&syscfg 0x2400>;
 		clocks          = <&rcc CK_KER_USB2PHY1>;
 		resets          = <&rcc USB2PHY1_R>;
-		vdd33-supply    = <&scmi_vdd33usb>;
 		status          = "disabled";
 	};
 
@@ -276,7 +271,6 @@
 		st,syscfg       = <&syscfg 0x2800>;
 		clocks          = <&rcc CK_KER_USB2PHY2EN>;
 		resets          = <&rcc USB2PHY2_R>;
-		vdd33-supply    = <&scmi_vdd33usb>;
 		status          = "disabled";
 	};
 
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -592,6 +592,10 @@
 		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
 		regulator-name = "vdd_emmc";
 	};
+	scmi_vdd3v3_usb: regulator@20 {
+		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+		regulator-name = "vdd3v3_usb";
+	};
 	scmi_v5v_hdmi: regulator@21 {
 		reg = <VOLTD_SCMI_STPMIC2_LDO5>;
 		regulator-name = "v5v_hdmi";
@@ -703,10 +707,12 @@
 };
 
 &usb2_phy1 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
 &usb2_phy2 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -636,6 +636,10 @@
 		reg = <VOLTD_SCMI_STPMIC2_LDO2>;
 		regulator-name = "vdd_emmc";
 	};
+	scmi_vdd3v3_usb: regulator@20 {
+		reg = <VOLTD_SCMI_STPMIC2_LDO4>;
+		regulator-name = "vdd3v3_usb";
+	};
 	scmi_vdd_sdcard: regulator@23 {
 		reg = <VOLTD_SCMI_STPMIC2_LDO7>;
 		regulator-name = "vdd_sdcard";
@@ -734,10 +738,12 @@
 };
 
 &usb2_phy1 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
 &usb2_phy2 {
+	vdd33-supply = <&scmi_vdd3v3_usb>;
 	status = "okay";
 };
 
